### PR TITLE
Merge/asoc v5.2

### PR DESCRIPTION
**Pull request by plbossart**
Now that SOF is mostly upstream, and some folks already submit SOF-related patches directly to alsa-devel, we should track upstream more closely instead of deriving our topic/sof-dev branch from a 5.1 release. This is an attempt to literally jump one integration step.
This doesn't change the fact that we are going to have to maintain backports, but instead of periodically merging the backports I suggest we periodically merge from Mark Brown's branch.

This branch is not even compile-tested but submitted to gather feedback on the approach and get CI results. 

I see very little review activity at the kernel level so I will take silence as consent. You've been warned.

### CODE REVIEW

**Comment by xiulipan** [comment]
@plbossart 
That was impossible to check all these commits I will go check all build errors and try to make some workaround if we are failing because of treating warnings as error.

File: drivers/acpi/utils.c [hunk 1]
```diff
@@ -846,9 +847,9 @@ acpi_dev_get_first_match_name(const char *hid, const char *uid, s64 hrv)
 	match.hrv = hrv;
 
 	dev = bus_find_device(&acpi_bus_type, NULL, &match, acpi_dev_match_cb);
-	return dev ? match.dev_name : NULL;
+	return dev ? match.adev : NULL;
```
Comment (xiulipan) [inline drivers/acpi/utils.c]:
```suggestion
	return dev ? match.name: NULL;
```

File: drivers/acpi/utils.c [hunk 2]
```diff
@@ -846,9 +847,9 @@ acpi_dev_get_first_match_name(const char *hid, const char *uid, s64 hrv)
...
 }
-EXPORT_SYMBOL(acpi_dev_get_first_match_name);
+EXPORT_SYMBOL(acpi_dev_get_first_match_dev);
```
Comment (xiulipan) [inline drivers/acpi/utils.c]:
```suggestion
EXPORT_SYMBOL(acpi_dev_get_first_match_name);
```

**Comment by xiulipan** [comment]
@plbossart That's all I got. Please rebase or somehow apply the changes.
817b4d64da036f5559297a2fdb82b8b14f4ffdcd  was the idea state of `drivers/acpi/utils.c` in linux master
```
git diff HEAD..817b4d64da036f5559297a2fdb82b8b14f4ffdcd  drivers/acpi/utils.c
diff --git a/drivers/acpi/utils.c b/drivers/acpi/utils.c
index 6d291331ac7e..5a2bae2b6c3a 100644
--- a/drivers/acpi/utils.c
+++ b/drivers/acpi/utils.c
@@ -847,9 +847,9 @@ acpi_dev_get_first_match_name(const char *hid, const char *uid, s64 hrv)
        match.hrv = hrv;

        dev = bus_find_device(&acpi_bus_type, NULL, &match, acpi_dev_match_cb);
-       return dev ? match.adev : NULL;
+       return dev ? match.dev_name : NULL;
 }
-EXPORT_SYMBOL(acpi_dev_get_first_match_dev);
+EXPORT_SYMBOL(acpi_dev_get_first_match_name);

 /*
  * acpi_backlight= handling, this is done here rather then in video_detect.c

```

**Comment by plbossart** [comment]
thanks @xiulipan i'll work on the merge issues. I must have missed something.

**Comment by cujomalainey** [comment]
I'm seeing some build errors in simple-card. Will follow up with a patch describing the issues.

**Comment by cujomalainey** [comment]
@plbossart I would check your browsers local setting and system time, your comment was 2h in the future (or maybe github is a bit confused)

**Comment by plbossart** [comment]
On 5/8/19 1:19 PM, Curtis Malainey wrote:
> @plbossart <https://github.com/plbossart> I would check your browsers 
> local setting and system time, your comment was 2h in the future (or 
> maybe github is a bit confused)

I am 2 hours ahead of you :-)

**Comment by cujomalainey** [comment]
Maybe timezone then? Because in theory my comments shouldn't sneak ahead of yours. They should be in order posted

**Comment by plbossart** [comment]
> Maybe timezone then? Because in theory my comments shouldn't sneak ahead of yours. They should be in order posted

oh yeah, I see it now "plbossart commented an hour from now". GitHub invented time travel!

**Comment by cujomalainey** [comment]
Merge conflict that caused a function to be duplicated
```
From 060aeafe12e2ecbd2fa4664c95b308b75e7ea282 Mon Sep 17 00:00:00 2001
From: Curtis Malainey <cujomalainey@chromium.org>
Date: Wed, 8 May 2019 13:13:13 -0700
Subject: [PATCH 1/2] Fix merge conflict in simple card

---
 sound/soc/generic/simple-card.c | 47 ---------------------------------
 1 file changed, 47 deletions(-)

diff --git a/sound/soc/generic/simple-card.c b/sound/soc/generic/simple-card.c
index 1ceb2717436a..9b568f578bcd 100644
--- a/sound/soc/generic/simple-card.c
+++ b/sound/soc/generic/simple-card.c
@@ -335,53 +335,6 @@ static int simple_dai_link_of(struct asoc_simple_priv *priv,
 	of_node_put(plat);
 	of_node_put(node);
 
-	/* loop for all dai-link */
-	do {
-		struct asoc_simple_card_data adata;
-		struct device_node *codec;
-		struct device_node *np;
-		int num = of_get_child_count(node);
-
-		/* get codec */
-		codec = of_get_child_by_name(node, is_top ?
-					     PREFIX "codec" : "codec");
-		if (!codec) {
-			ret = -ENODEV;
-			goto error;
-		}
-
-		of_node_put(codec);
-
-		/* get convert-xxx property */
-		memset(&adata, 0, sizeof(adata));
-		for_each_child_of_node(node, np)
-			simple_get_conversion(dev, np, &adata);
-
-		/* loop for all CPU/Codec node */
-		for_each_child_of_node(node, np) {
-			/*
-			 * It is DPCM
-			 * if it has many CPUs,
-			 * or has convert-xxx property
-			 */
-			if (num > 2 ||
-			    adata.convert_rate || adata.convert_channels)
-				ret = func_dpcm(priv, np, codec, li, is_top);
-			/* else normal sound */
-			else
-				ret = func_noml(priv, np, codec, li, is_top);
-
-			if (ret < 0) {
-				of_node_put(np);
-				goto error;
-			}
-		}
-
-		node = of_get_next_child(top, node);
-	} while (!is_top && node);
-
- error:
-	of_node_put(node);
 	return ret;
 }
 
-- 
2.21.0.1020.gf2820cf01a-goog

```
and partially implemented patch 
```
From 3a361c7598fed0416e8784cef6de851e7cdae2f9 Mon Sep 17 00:00:00 2001
From: Curtis Malainey <cujomalainey@chromium.org>
Date: Wed, 8 May 2019 13:18:00 -0700
Subject: [PATCH 2/2] Fix merge conflict in rcar/src.c

---
 sound/soc/sh/rcar/src.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/sound/soc/sh/rcar/src.c b/sound/soc/sh/rcar/src.c
index cd98fc764744..c0c34cc99756 100644
--- a/sound/soc/sh/rcar/src.c
+++ b/sound/soc/sh/rcar/src.c
@@ -195,7 +195,6 @@ static void rsnd_src_set_convert_rate(struct rsnd_dai_stream *io,
 	struct rsnd_priv *priv = rsnd_mod_to_priv(mod);
 	struct device *dev = rsnd_priv_to_dev(priv);
 	struct snd_pcm_runtime *runtime = rsnd_io_to_runtime(io);
-	const struct soc_device_attribute *soc = soc_device_match(ov_soc);
 	int is_play = rsnd_io_is_play(io);
 	int use_src = 0;
 	u32 fin, fout;
-- 
2.21.0.1020.gf2820cf01a-goog

```

**Comment by cujomalainey** [comment]
Also sof/compressed.c doesn't compile, but I think that might be caused by topic/sof-dev

**Comment by plbossart** [comment]
this was a major fail. I will first remove everything that is not upstream and conflicts, so that those merges are easier to do. I had too many manual things to fix and it didn't go well at all.

**Comment by plbossart** [comment]
Take 2 updates ready.

List of files different from asoc-v5.2:

````
A       .travis.yml
A       CODEOWNERS
M       drivers/acpi/blacklist.c
M       include/sound/sof/dai.h
M       include/uapi/sound/sof/abi.h
D       include/uapi/sound/sof/eq.h
D       include/uapi/sound/sof/manifest.h
D       include/uapi/sound/sof/tone.h
D       include/uapi/sound/sof/trace.h
M       sound/pci/hda/hda_codec.c
M       sound/soc/codecs/hdac_hda.c
M       sound/soc/intel/Kconfig
M       sound/soc/intel/boards/Kconfig
M       sound/soc/intel/boards/Makefile
M       sound/soc/intel/boards/bxt_da7219_max98357a.c
A       sound/soc/intel/boards/bxt_pcm512x.c
A       sound/soc/intel/boards/bxt_wm8804.c
M       sound/soc/sof/Kconfig
M       sound/soc/sof/Makefile
M       sound/soc/sof/core.c
M       sound/soc/sof/intel/Kconfig
M       sound/soc/sof/intel/Makefile
M       sound/soc/sof/intel/cnl.c
M       sound/soc/sof/intel/hda.h
M       sound/soc/sof/ipc.c
M       sound/soc/sof/loader.c
M       sound/soc/sof/ops.c
M       sound/soc/sof/pcm.c
M       sound/soc/sof/sof-pci-dev.c
M       sound/soc/sof/xtensa/Makefile
````

and list of patches not in asoc-v5.2 (the first two were picked up by Takashi)

````
ALSA: hda: fix unregister device twice on ASoC driver
ASoC: hdac_hda: overwrite hdev type to HDA_DEV_ASOC
ASoC: SOF: soundwire: add initial soundwire support
ASoC: SOF: fix race in FW boot timeout handling
ASoC: SOF: fix error in verbose ipc command parsing
ASoC: SOF: pcm: remove runtime PM calls during pcm open/close
ASoC: SOF: Intel: ICL: add Icelake SSP count
ASoC: SOF: Intel: ICL add Icelake chip info struct
ASoC: SOF: uapi: remove unused sof header files
ASoC: Intel: bxt-da7219-max98357a: add BE dailink for dmic16k
ASoC: Intel: add machine driver for BXT/APL with pcm512x codec
ASoC: Intel: boards: initial support for Hifiberry Digiplus w/ WM8804
ASoC: Intel: Kconfig: disable SST and legacy drivers when SOF is selected
acpi: blacklist: remove quirk for Dell XPS13 when SOF is enabled
ASoC: SOF: enable DEBUG by default
[NOT FOR UPSTREAM] ASoC: Intel: ApolloLake support for SOF
[NOT FOR UPSTREAM] CODEOWNERS: Initial version
````

**Comment by cujomalainey** [comment]
Passes a sanity make allyesconfig build

**Comment by plbossart** [comment]
I will take silence from the Intel SOF team as approval.
Let's merge.

**Pull request merged**